sympy/matrices/matrices.py

```diff
@@ -4260,6 +4260,101 @@ def QRsolve(self, b):
             x.append(tmp / R[j, j])
         return self._new([row._mat for row in reversed(x)])
 
+    def rank_decomposition(self, iszerofunc=_iszero, simplify=False):
+        r"""Returns a pair of matrices (`C`, `F`) with matching rank
+        such that `A = C \cdot F`.
+
+        Parameters
+        ==========
+
+        iszerofunc : Function, optional
+            A function used for detecting whether an element can
+            act as a pivot.  ``lambda x: x.is_zero`` is used by default.
+
+        simplify : Bool or Function, optional
+            A function used to simplify elements when looking for a
+            pivot. By default SymPy's ``simplify`` is used.
+
+        Returns
+        =======
+
+        (C, F) : Matrices
+            `F` is the reduced row echelon form (RREF) of `A`, and `C`
```

Comment:
`F` is the RREF without the zero rows. We can be more specific as well: that `C`'s columns are a subset of `A`'s columns. Though, if we say this, we're pinning down the exact functionality. Your tests don't verify this.

It might be better to say `C` and `F` are full-rank matrices (with rank the same as `A`) whose product gives `A`.

Reply:
It says that the columns are a subset of As in the Notes, but I agree that although the notes can say this, it need not (perhaps) be a requirement of the return value. I like your suggestion about saying they are simply full-rank matrices.

Reply:
Well, I agree this can be more valid description after the method using singular value decomposition added.
Though, there is still no SVD in sympy.